pkg/utils/common.go

```
@@ -252,3 +253,44 @@ func GetUpgradeTempDir(config *v1.RunConfig) string {
 	}
 	return filepath.Join("/", "tmp", "elemental-upgrade")
 }
+
+// IsLocalUrl returns true if the url has no scheme or it has "file" scheme
+// and returns false otherwise. Error is not nil if the url can't be parsed.
```

why aren't we returning a proper err if this errors out? We cant be inconsistent, either there is an error and we return it or we don't, otherwise we need to be checking the error and the return value in order to know what happened no? Error out if there was an error IMO.

---

Yes, this is also an option, however you can't distinguish invalid URLs from remote ones in that case. I thought this could be also eventually used to sanitize URLs and/or tests remote urls. But yes, we can eventually have a IsLocalURl method and if needed another IsRemoteUrl method.

---

Note you can ignore the error with this method, you could eventually use:

```golang
if IsLocalUrl("someurl") {
    ...
}
```
It only returns true if the URL is local. If it returns false it is not a local url and it is eventually an invalid URL if err is not nil.